components/cdi/weld/src/main/java/com/kumuluz/ee/configuration/cdi/interceptors/ConfigBundleInterceptor.java

```diff
@@ -62,96 +66,186 @@ public Object loadConfiguration(InvocationContext ic) throws Exception {
+                    if (watchAllFields || (fieldAnnotation != null && fieldAnnotation.watch())) {
+                        deployWatcher(target, method, getKeyName(targetClass, method.getName(), keyPrefix));
+                    }
 
-                    Optional<Double> value = configurationUtil.getDouble(getKeyName(targetClass, m.getName()));
+                    // process nested objeccts
+                } else if (!parameterType.isArray()) {
 
-                    if (value.isPresent()) {
-                        m.invoke(target, value.get());
-                    }
+                    processedClassRelations.put(targetClass, parameterType);
 
-                    deployWatcher(target, targetClass, m, getKeyName(targetClass, m.getName()));
+                    Object nestedTarget = processNestedObject(targetClass, method, parameterType, keyPrefix,
+                            processedClassRelations, -1, watchNestedClass);
 
-                } else if (m.getParameters()[0].getType().equals(Integer.class)) {
+                    // invoke setter method with initialised instance
+                    method.invoke(target, nestedTarget);
 
-                    Optional<Integer> value = configurationUtil.getInteger(getKeyName(targetClass, m.getName()));
+                    // process arrays
+                } else {
 
-                    if (value.isPresent()) {
-                        m.invoke(target, value.get());
-                    }
+                    Class componentType = parameterType.getComponentType();
 
-                    deployWatcher(target, targetClass, m, getKeyName(targetClass, m.getName()));
+                    Object array = Array.newInstance(componentType, configurationUtil.getListSize(getKeyName
+                            (targetClass, method.getName(), keyPrefix)).orElse(0));
 
-                } else if (m.getParameters()[0].getType().equals(Long.class)) {
+                    // process list of primitives
+                    if (Arrays.asList(primitives).contains(componentType)) {
+                        for (int i = 0; i < Array.getLength(array); i++) {
+                            Array.set(array, i, getValueOfPrimitive(componentType, getKeyName(targetClass, method
+                                    .getName(), keyPrefix) + "[" + i + "]").get());
+                        }
 
-                    Optional<Long> value = configurationUtil.getLong(getKeyName(targetClass, m.getName()));
+                        // process list of nested classes
+                    } else {
+                        for (int i = 0; i < Array.getLength(array); i++) {
+                            Array.set(array, i, processNestedObject(targetClass, method, componentType, keyPrefix,
+                                    processedClassRelations, i, watchNestedClass));
 
-                    if (value.isPresent()) {
-                        m.invoke(target, value.get());
+                        }
```

Comment:
Same as for the nested object. If no keys exist, leave it as is.

Reply:
Fixed.